autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/ui/credential/management/viewing/AutofillManagementStringBuilder.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.autofill.impl.ui.credential.management.viewing
+
+import android.content.Context
+import android.content.res.Resources
+import com.duckduckgo.autofill.impl.R
+import com.duckduckgo.di.scopes.FragmentScope
+import com.duckduckgo.sync.api.DeviceSyncState
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+
+interface AutofillManagementStringBuilder {
+    fun stringForDeleteLoginsConfirmationDialogMessage(numberToDelete: Int): String
+    fun stringForDeleteAllPasswordsDialogConfirmationDialogTitle(numberToDelete: Int): String
+}
+
+@ContributesBinding(FragmentScope::class)
+class AutofillManagementStringBuilderImpl @Inject constructor(
+    private val context: Context,
+    private val deviceSyncState: DeviceSyncState,
+) : AutofillManagementStringBuilder {
+
+    override fun stringForDeleteAllPasswordsDialogConfirmationDialogTitle(numberToDelete: Int): String {
+        return context.resources.getQuantityString(
+            R.plurals.credentialManagementDeleteAllPasswordsDialogConfirmationTitle,
+            numberToDelete,
+            numberToDelete,
+        )
+    }
+
+    override fun stringForDeleteLoginsConfirmationDialogMessage(numberToDelete: Int): String {
+        with(context.resources) {
+            val firstMessage = deleteAllPasswordsWarning(numberToDelete)
```

Comment:
ok, now I'm a bit confused. Do we use the display the same warning when removing all vs when removing only 1?

There's also a discrepancy in the method names not sure if intended: one says Passwords, the other one says Logins. Should be logins in both?

Reply:
_>Do we use the display the same warning when removing all vs when removing only 1?_

**Dialog Title**
We are re-using the same code here because there are a few scenarios:

1. deleting a single password
1. choosing to delete-all when you have a single password saved
1. choosing to delete-all when you have multiple passwords saved

For 1. and 2., we show the same title. For 3, it's a different one.


**Dialog Message**

For messages, we change the string depending on whether:

- sync is enabled or not
- whether deleting one or multiple (pluralisation of some words change)


**Making the code clearer**
As discussed offline, will rename the string builder functions so that: 
- remove the `all` part from the function name (since it's only sometimes dealing with delete-all vs deleting one)
- naming is consistent; both referring to passwords rather than login